.azure-pipelines/sync-fork.yml

```diff
@@ -0,0 +1,81 @@
+#sync-fork.yml
+# To sync fork with the upstream repo and this is done on a schedule at 10 pm PST daily.To manually do the sync we follow the below steps 
+# and the same has been incorporated in a pipeline to make it automated.
+#Input params: we provide the fork repo url and the upstream repo url defined at pipeline definition and their branches to be synced
+# 1. git clone github.com/microsoft/x
+# 2.cd x
+# 3.git checkout main (our main branch)
+# 4.git remote add upstream github.com/upstream-org/x
+# 5.git fetch upstream
+# 6.git merge -ff only upstream/master (into our main branch)
+# 7.git push
+# $PASSKEY: Pipeline variable defined for PAT token that is saved as a secret in pipeline definition variable.
+
+trigger: none
+
+schedules:
+  # At 5am UTC (10pm PST) daily
+  - cron: "0 5 * * *"
+    displayName: Sync
+    always: false # only run if there has been code changes to the jdk codebase
+    branches:
+      include:
+        - "*"
+
+stages:
+- stage: sync
+  displayName: "Mirror upstream repo"
+  jobs:
+  - job: sync
+    displayName: "sync repository with upstream"
+
+    pool:
+      name: JEG-ubuntu20.04-x64-EO
+    
+    steps:     
+      - bash: |
+          git config --global --add http.https://dev.azure.com.extraheader "AUTHORIZATION: bearer $(System.AccessToken)"
+          git config --global core.autocrlf input
+          git config --global user.email "javaplatinfra@microsoft.com"
+          git config --global user.name "JEG- $(Build.DefinitionName)"
+          # git config --global user.email "javaplatinfra@microsoft.com"
+          # git config --global user.name "JEG- $(Build.DefinitionName)"
+        displayName: "git config" 
+
+      - bash: |
+          # ${parameter/#pattern/string} If pattern is preceded by ‘#’ (the third form above), 
+          # it must match at the beginning of the expanded value of parameter
+          fork_repo_url=$(FORK_REPO_URL)
+          repourl="${fork_repo_url#https://github.com/schandra13/}"
+          echo "reponame is : $repourl"
+          echo "##vso[task.setvariable variable=FORK_FOLDER]$repourl"
+        displayName: "Fetch fork repo folder name"
+  
+      - bash: |
+          git clone $(FORK_REPO_URL)
+        displayName: "Clone fork repository"
+      
+      - bash: |
+          echo "Fork_Folder is : $(FORK_FOLDER)"
+          cd $(FORK_FOLDER)
+          git checkout $(FORK_BRANCH)
+        displayName: "Change directory to fork folder and checkout fork branch"     
+
+      - bash: |
+          cd $(FORK_FOLDER)
+          git remote add upstream $(UPSTREAM_REPO_URL)          
+          git fetch upstream
```

Comment:
Probably a good idea to stop here, and make the actual merge it's own task. (The fetch _shouldn't fail_ of course, but it would be very useful to know when it ever does!).

Reply:
Moved merge to a separate task